misc/scripts/generate-code-scanning-query-list.py

```diff
@@ -0,0 +1,156 @@
+import subprocess
+import json
+import csv
+import sys
+import os
+
+"""
+This script collects CodeQL queries that are part of code scanning query packs
+and prints CSV data to stdout that describes which packs contain which queries.
+
+Errors are printed to stderr. This script requires that 'git' and 'codeql' commands
+are on the PATH. It'll try to automatically set the CodeQL search path correctly,
+as long as you run the script from one of the following locations:
+ - anywhere from within a clone of the CodeQL Git repo
+ - from the parent directory of a clone of the CodeQL Git repo (assuming 'codeql' 
+   and 'codeql-go' directories both exist)
+"""
+
+# Define which languages and query packs to consider
+languages = [ "cpp", "csharp", "go", "java", "javascript", "python"]
+packs = [ "code-scanning", "security-and-quality", "security-extended" ]
+
+# Define CodeQL search path so it'll find the CodeQL repositories:
+#  - anywhere in the current Git clone (including current working directory)
+#  - the 'codeql' subdirectory of the cwd
+#
+# (and assumes the codeql-go repo is in a similar location)
+codeql_search_path = "./codeql:./codeql-go:"   # will be extended further down
```

Comment:
@hmakholm may be able to advise whether you actually need to set the search path, if the CLI and the query repos are in the right sibling directories.

Reply:
I tried that, but that didn't seem to work

Reply:
The unzipping instruction puts the `codeql` binary as `codeql-cli/codeql/codeql` due to the clash between the `codeql` directory prefix inside `codeql-linux64.zip` and the name of the `github/codeql` repository. So the usual sibling heuristic will indeed not work -- the relevant directories are uncles not siblings.

For use in automation I think it does feel safer to build a search path explicitly than to rely on default "action at a distance" heuristics, so :+1: from me.